patroni/postgresql/__init__.py

```diff
@@ -804,38 +812,32 @@ def remove_data_directory(self):
             logger.exception('Could not remove data directory %s', self._data_dir)
             self.move_data_directory()
 
-    def pick_synchronous_standby(self, cluster):
-        """Finds the best candidate to be the synchronous standby.
+    def current_sync_state(self, cluster):
+        """Returns current synchronous replication state as a dict:
+        * numsync: number of synchronous nodes requested(including leader)
+        * sync: set of nodes potentially being synced to
+        * active: set of nodes that are sync capable
+         """
+        sync_standby_names = self.query("SHOW synchronous_standby_names").fetchone()[0]
+        ssn_data = parse_sync_standby_names(sync_standby_names)
 
-        Current synchronous standby is always preferred, unless it has disconnected or does not want to be a
-        synchronous standby any longer.
-
-        :returns tuple of candidate name or None, and bool showing if the member is the active synchronous standby.
-        """
-        current = cluster.sync.sync_standby
-        current = current.lower() if current else current
+        active = [self.name]
         members = {m.name.lower(): m for m in cluster.members}
-        candidates = []
-        # Pick candidates based on who has flushed WAL farthest.
-        # TODO: for synchronous_commit = remote_write we actually want to order on write_location
+
         for app_name, state, sync_state in self.query(
                 "SELECT pg_catalog.lower(application_name), state, sync_state"
                 " FROM pg_catalog.pg_stat_replication"
                 " ORDER BY flush_{0} DESC".format(self.lsn_name)):
             member = members.get(app_name)
             if state != 'streaming' or not member or member.tags.get('nosync', False):
                 continue
-            if sync_state == 'sync':
-                return member.name, True
-            if sync_state == 'potential' and app_name == current:
-                # Prefer current even if not the best one any more to avoid indecisivness and spurious swaps.
-                return cluster.sync.sync_standby, False
-            if sync_state in ('async', 'potential'):
-                candidates.append(member.name)
-
-        if candidates:
-            return candidates[0], False
-        return None, False
+            active.append(member.name)
```

Comment:
Just to double check: a standby with `pg_stat_replication.sync_state  in ('async', 'potential'):`  is still considered `active` for the purposes of `QuorumStateResolver`?